docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/apidesc/DockerBuildDescribable.java

```diff
@@ -0,0 +1,192 @@
+package com.nirima.jenkins.plugins.docker.apidesc;
+
+import com.github.dockerjava.api.command.BuildImageCmd;
+import hudson.Extension;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import jenkins.model.Jenkins;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.CheckForNull;
+import java.io.Serializable;
+import java.net.URI;
+
+/**
+ * Plain Docker Build Describable. For splitting API wrapping and jenkins specific features.
+ *
+ * @author Kanstantsin Shautsou
+ */
+public class DockerBuildDescribable implements Describable<DockerBuildDescribable>, Serializable {
+
+//    /**
+//     *  A Git repository URI or HTTP/HTTPS context URI. If the URI points to a single text file,
+//     *  the file’s contents are placed into a file called Dockerfile and the image is built from that file.
+//     *  If the URI points to a tarball, the file is downloaded by the daemon and the contents
+//     *  therein used as the context for the build. If the URI points to a tarball and the dockerfile
+//     *  parameter is also specified, there must be a file with the corresponding path inside the tarball.
+//     */
+//    @CheckForNull
+//    private URI remote;
+
+    /**
+     * Suppress verbose build output. API
+     */
+    private boolean q;
+
+    /**
+     * Do not use the cache when building the image. API
+     */
+    private boolean nocache;
+
+    /**
+     * Attempt to pull the image even if an older image exists locally. API
+     */
+    private boolean pull;
+
+    /**
+     * Remove intermediate containers after a successful build (default behavior). API
+     */
+    private boolean rm = true;
+
+//    /**
+//     *  Always remove intermediate containers (includes rm). API
+//     */
+//    private boolean forcerm;
+//
+//    /**
+//     *  Set memory limit for build. API
+//     */
+//    private long memory;
+//
+//    /**
+//     * Total memory (memory + swap), -1 to disable swap. API
+//     */
+//    private long memswap;
+//
+//    /**
+//     * CPU shares (relative weight).
+//     */
+//    private long cpushares;
+
+    @DataBoundConstructor
+    public DockerBuildDescribable() {
+    }
+
+//
+//    /**
+//     *  CPUs in which to allow execution (e.g., 0-3, 0,1).
+//     */
+//    @CheckForNull
+//    private String cpusetcpus;
+//
+//    public URI getRemote() {
+//        return remote;
+//    }
+//
+//    public void setRemote(URI remote) {
+//        this.remote = remote;
+//    }
+
+    public boolean isQ() {
+        return q;
+    }
+
+    @DataBoundSetter
+    public void setQ(boolean q) {
```

Comment:
F*cking variable naming in docker API. Just to have less chaos following their upstream naming either it will be hard for me to sync API->docker-java&docker-plugin 
https://github.com/docker-java/docker-java/pull/299
https://github.com/docker-java/docker-java/pull/299/files#diff-50e3dbf43281453d0708f5249573aae6R16
https://docs.docker.com/reference/api/docker_remote_api_v1.20/#build-image-from-a-dockerfile